src/icon-server/fastify-plugin.js

```diff
@@ -0,0 +1,64 @@
+import { Type as T } from '@sinclair/typebox'
+import fp from 'fastify-plugin'
+import { kIconApi } from '../mapeo-project.js'
+
+export default fp(iconServerPlugin, {
+  fastify: '4.x',
+  name: 'mapeo-icon-server',
+})
+
+// iconDocId is a hex encoded 32-byte string
+const HEX_REGEX_32_BYTES = '^[0-9a-fA-F]{64}$'
+// projectId is encoded to a z-base-32 52-character string (32 bytes)
+const Z_BASE_32_REGEX_32_BYTES = '^[0-9a-zA-Z]{52}$'
+const ICON_DOC_ID_STRING = T.String({ pattern: HEX_REGEX_32_BYTES })
+const PROJECT_ID_STRING = T.String({ pattern: Z_BASE_32_REGEX_32_BYTES })
+
+const PARAMS_JSON_SCHEMA = T.Object({
+  iconDocId: ICON_DOC_ID_STRING,
+  projectId: PROJECT_ID_STRING,
+  size: T.String(),
+  pixelDensity: T.Number(),
+})
+
+/**
+ * @typedef {Object} IconServerPluginOpts
+ * @property {import('fastify').RegisterOptions['prefix']} prefix
+ **/
+
+/** @type {import('fastify').FastifyPluginAsync<import('fastify').RegisterOptions & IconServerPluginOpts>} */
+async function iconServerPlugin(fastify, options) {
+  fastify.register(routes, options)
+}
+
+/**
+ * @typedef {(projectId: string) =>
+ * Promise<import('../mapeo-project.js').MapeoProject>} GetProject
+ */
+
+/** @type {import('fastify').FastifyPluginAsync<
+ * Omit<IconServerPluginOpts, 'prefix'>,
+ * import('fastify').RawServerDefault,
+ * import('@fastify/type-provider-typebox').TypeBoxTypeProvider>}
+ * */
+async function routes(fastify) {
+  const { getProject } = fastify
```

Comment:
Type error here happens because I should somehow tell typescript that `fastify` param here is decorated with the `getProject` method. 
I tried a couple of things but couldn't make it work...
@gmaclennan ideas??